services/heartbeat/types.go

```diff
@@ -23,16 +23,17 @@ package heartbeat
 import (
 	"time"
 
+	"github.com/m3db/m3cluster/services"
 	"github.com/m3db/m3x/watch"
 )
 
 // Store is a heartbeat store
 type Store interface {
 	// Heartbeat sends heartbeat for a service instance with a ttl
-	Heartbeat(service, instance string, ttl time.Duration) error
+	Heartbeat(service string, instance services.PlacementInstance, ttl time.Duration) error
 
 	// Get gets healthy instances for a service
-	Get(service string) ([]string, error)
+	Get(service string) ([]services.PlacementInstance, error)
```

Comment:
I like the idea of having a more lightweight `Get` function that only returns the IDs. As for the watch question, I've been going back and forth on this while trying to predict the various use cases. I think what I'm going to do is keep the `heartbeat.Store.Watch` function as returning just the string IDs, but add a `WatchInstances` function that will return a watch that deserializes the values. Then I'll add a `WithInstances(bool)` function or similar to the `services.QueryOptions` interface, and that will determine which of the two `store.Watch*` functions is used when the caller calls `services.Services.Watch(...)`